### ISSUE TITLE
Using torch.cross without specifying the dim arg is deprecated, use torch.linalg.cross

### ISSUE THREAD

**Issue by fnachon** [YaoYinYing/RFdiffusion#6]
I found a warning with pytorch 2.3.1 

```
/Users/florian/miniconda3/envs/rfdiffusion/lib/python3.10/site-packages/rfdiffusion/util.py:253: UserWarning: Using torch.cross without specifying the dim arg is deprecated.
Please either pass the dim explicitly or simply use torch.linalg.cross.
The default value of dim will change to agree with that of linalg.cross in a future release. (Triggered internally at /Users/runner/miniforge3/conda-bld/libtorch_1719361051023/work/aten/src/ATen/native/Cross.cpp:66.)
  Z = torch.cross(Xn, Yn)
```

I fixed it by replacing `    Z = torch.cross(Xn, Yn)` by `    Z = torch.linalg.cross(Xn, Yn)` in `.../site-packages/rfdiffusion/util.py`

Here is my full working conda environment:
[rfdiffusion_mac_env.txt](https://github.com/user-attachments/files/19406169/rfdiffusion_mac_env.txt)

**Comment by fnachon** [comment]
Same issue in rfdiffusion/util_module.py with `CBrotaxis1 = (CBr-CAr).cross(NCr-CAr)` and `CBrotaxis2 = (CBr-CAr).cross(NCpp)`

fixed code (line 254)
```
        # CB bend
        basexyzs = self.xyzs_in_base_frame[seq]
        NCr = 0.5*(basexyzs[:,:,2,:3]+basexyzs[:,:,0,:3])
        CAr = (basexyzs[:,:,1,:3])
        CBr = (basexyzs[:,:,4,:3])
        #CBrotaxis1 = (CBr-CAr).cross(NCr-CAr)
        CBrotaxis1 = torch.linalg.cross(CBr-CAr, NCr-CAr)
        CBrotaxis1 /= torch.linalg.norm(CBrotaxis1, dim=-1, keepdim=True)+1e-8
        
        # CB twist
        NCp = basexyzs[:,:,2,:3] - basexyzs[:,:,0,:3]
        NCpp = NCp - torch.sum(NCp*NCr, dim=-1, keepdim=True)/ torch.sum(NCr*NCr, dim=-1, keepdim=True) * NCr
        #CBrotaxis2 = (CBr-CAr).cross(NCpp)
        CBrotaxis2 = torch.linalg.cross(CBr-CAr, NCpp)
        CBrotaxis2 /= torch.linalg.norm(CBrotaxis2, dim=-1, keepdim=True)+1e-8
```
